hack/test/basic-integration.sh

```diff
@@ -32,9 +32,41 @@ esac
 
 mkdir -p "${TMP}"
 
-"${OSCTL}" cluster create --name basic-integration --image "${TALOS_IMG}" --masters=3 --mtu 1440 --cpus 4.0 --wait --endpoint "${ENDPOINT}"
+case ${PROVISIONER} in
+  docker)
+    "${OSCTL}" cluster create \
+      --provisioner docker \
+      --image "${TALOS_IMG}" \
+      --name basic-integration \
+      --masters=3 \
+      --mtu 1500 \
+      --memory 2048 \
+      --cpus 4.0 \
+      --wait \
+      --endpoint "${ENDPOINT}"
+    ;;
+
+  firecracker)
+    "${OSCTL}" cluster create \
+      --provisioner firecracker \
+      --name basic-integration \
+      --masters=3 \
+      --mtu 1500 \
+      --memory 2048 \
+      --cpus 2.0 \
+      --cidr 172.20.0.0/24 \
+      --init-node-as-endpoint \
+      --wait \
+      --install-image docker.io/autonomy/installer:latest
+    ;;
+
+  *)
+    echo "unknown provisioner: ${PROVISIONER}"
+    exit 1
+    ;;
+esac
 
-"${INTEGRATION_TEST}" -test.v -talos.osctlpath "${OSCTL}" -talos.k8sendpoint "${ENDPOINT}:6443"
```

Comment:
iirc, `talos.k8sendpoint` is still required for Docker, because internal endpoint (10.5.0.2) is different from external (`docker` in our case)

Reply:
for Firecracker we don't need, as init node is routable both externally and internally

Reply:
but if we define `ENDPOINT=172.20.0.2` for Firecracker, it will be no-op

Reply:
ahh thanks, was banging my head against the wall